src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -46,18 +55,44 @@ class KubeTransferStrategy implements TransferStrategy {
     @Inject
     private K8sService k8sService
 
+    @Inject
+    private CleanupStrategy cleanup
+
+    @Inject
+    @Named(TaskExecutors.IO)
+    private ExecutorService executor
+
     @Override
     BlobCacheInfo transfer(BlobCacheInfo info, List<String> command) {
-        final podName = podName(info)
-        final pod = k8sService.transferContainer(podName, blobConfig.s5Image, command, blobConfig)
-        final terminated = k8sService.waitPod(pod, blobConfig.transferTimeout.toMillis())
-        final stdout = k8sService.logsPod(podName)
-        return terminated
+        final name = getName(info)
+        final job = k8sService.transferJob(name, blobConfig.s5Image, command, blobConfig)
+        final podList = k8sService.waitJob(job, blobConfig.transferTimeout.toMillis())
+        final size = podList.items.size()
+
+        if( size < 1 )
+            throw new TransferTimeoutException("Transfer job timed out")
+
+        // Find the latest created pod among the pods associated with the job
+        final latestPod = K8sHelper.findLatestPod(podList)
+
+        final podName = latestPod.metadata.name
+        final pod = k8sService.getPod(podName)
+        final terminated = k8sService.waitPod(pod, name, blobConfig.transferTimeout.toMillis())
+        final stdout = k8sService.logsPod(podName, name)
+
+        final result = terminated
                 ? info.completed(terminated.exitCode, stdout)
                 : info.failed(stdout)
+
+        // delete job
+        if( cleanup.shouldCleanup(terminated.exitCode) ) {
+            CompletableFuture.supplyAsync (() -> k8sService.deleteJob(job.metadata.name), executor)
```

Comment:
Should not be added also a delete for the pod? also worth adding a check that `metadata` and `name` are not null

Reply:
If we delete the job, pods associated with it also get deleted automatically

Reply:
done